### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by LOctavio**
For this milestone I did the following:

- Create a class Classroom with the following:
  - @label instance variable, should be initialized in the constructor.
  - Setter and getter for @label.
- Create the has-many/belongs-to relationship between Classroom and Student. The following was implemented:
  - Create the has-many side (a classroom has many students).
  - Create the belongs-to side (a student belongs to a classroom).
- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables.
- Create a class Rental with the following:
  - @date instance variable, should be initialized in the constructor.
  - Setter and getter for @date.
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following was implemented:
  - Create the has-many side of Book and Rental (a book has many rentals).
  - Create the belongs-to side of Rental and Book (a rental belongs to a book).
  - Create the has-many side of Person and Rental (a person has many rentals).
  - Create the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students.push(student)
+  end
```
Comment (Davidosky007) [inline classroom.rb]: * I suggest you set each student to classroom in this method for example `student.classroom = self` this helps the association work both ways `has-many` and `belongs-to` so for every student created a classroom is assigned to it and vice versa. [Refer to this article on association](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
-  attr_accessor :age, :name
+  attr_accessor :age, :name, :rental
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
     @corrector = Corrector.new
+    @rental = []
```
Comment (Davidosky007) [inline person.rb:13]: * Kindly create a method for adding rentals in `book.rb` which has an association of `has-many-rentals` you can do same for `person.rb` as it also have the same association of `has-many-rentals` 
image
## [Refer to this article for a refresher on associations](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rental
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
```
Comment (Davidosky007) [inline book.rb:7]: * Kindly create a method for adding rentals in `book.rb` which has an association of `has-many-rentals` you can do same for `person.rb` as it also have the same association of `has-many-rentals` 
image
## [Refer to this article for a refresher on associations](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @LOctavio 👋, 

Good job so far! 💪 
* Descriptive commit messages ✔️ 
* Well written PR ✔️ 
* No linter errors ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
* I think you can make your readme file more descriptive ie modifying the project name, authors link, setup etc 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🎉✨**
Hello @LOctavio 👋🏽
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**